### PR TITLE
[PLAY-580] Maplibre Map POC

### PR DESCRIPTION
**Pull request by nidaqg**
Using [MapLibre](https://maplibre.org/) for the Map kit doc examples

#### Screens

map-kit

### CODE REVIEW

File: playbook-website/app/javascript/site_styles/map_styles/_map_default.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@import "playbook-ui/dist/tokens/colors";
+@import "playbook-ui/dist/tokens/spacing";
+@import "playbook-ui/dist/tokens/shadows";
+@import url("https://unpkg.com/maplibre-gl@2.4.0/dist/maplibre-gl.css");
+@import url("https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-draw/v1.3.0/mapbox-gl-draw.css");
+
+
+.pb_map {
+  height: 600px;
```
Comment (jasperfurniss) [inline playbook-website/app/javascript/site_styles/map_styles/_map_default.scss:9]: I just wonder how this should function on mobile/responsive views... 
Height at 600px always is pretty strict. Any thoughts? @nidaqg

File: playbook/app/pb_kits/playbook/pb_map/docs/_map_default.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            style: 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
+            center: [-75.379143, 39.831200],
+            zoom: 13,
+        })
+        //set marker/pin
+        /* eslint-disable-next-line */
+        const marker = new maplibregl.Marker({
+          color: "#0056CF",
+        }).setLngLat([-75.379143, 39.831200])
+        .setPopup(new maplibregl.Popup({className: 'map_popup', closeButton: false}).setHTML(`<h4 class="pb_title_kit_size_4">Hello World!</h4>`)) // add popup
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_map/docs/_map_default.jsx:25]: Let's extract this string into it's own component above....and then we can actually use our kit! I can help if you need more direction on this.

File: playbook/app/pb_kits/playbook/pb_map/docs/_map_default.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        } else {
+         const map = new maplibregl.Map({
+            container: mapContainerRef.current,
+            style: 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
+            center: [-75.379143, 39.831200],
+            zoom: 13,
+        })
+        //set marker/pin
+        /* eslint-disable-next-line */
+        const marker = new maplibregl.Marker({
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_map/docs/_map_default.jsx:22]: You might even want to separate this out a little into it's own object. That will make this useEffect a little more readable.

File: playbook-website/app/javascript/site_styles/map_styles/_map_default.scss (hunk 1 above)
Comment (nidaqg) [reply playbook-website/app/javascript/site_styles/map_styles/_map_default.scss:9]: The library needs a height passed to it for the map to render. we can also set a height to the parent div and set this height to 100%,  I can try it with max-height if you want

**Pull request merged**